pkg/ruler/api.go

```diff
@@ -154,6 +155,14 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	typ := strings.ToLower(req.URL.Query().Get("type"))
+	if typ != "" && typ != "alert" && typ != "record" {
+		respondError(logger, w, fmt.Sprintf("not supported value %q", typ))
+		return
+	}
+	returnAlerts := typ == "" || typ == "alert"
+	returnRecording := typ == "" || typ == "record"
+
 	w.Header().Set("Content-Type", "application/json")
 	rgs, err := a.ruler.GetRules(req.Context())
```

Comment:
I'd consider adding this filter directly to the proto request. 

On each API call, we execute a fan-out request to each ruler that owns rule groups for the tenant - even with the filtering at an API level, we'll still take the network cost by bringing all the information we'll eventually filter out.

Now, it's worth considering whether clients actually use this filter, and I think the answer is no (For the time being)? So perhaps it doesn't matter all too much... For now. WDYT @pracucci?

Reply:
Sure, I don't like the network overhead also.
I pushed the filter down to the protobuf request, would you PTAL if that is what you had in mind?
Also, I'm not sure if the change in protobuf is breaking, since the filter is required. I considered it to be optional, but that seems to be still somehow experimental and requires to be allowed by a flag.